entities/pool.go

```diff
@@ -415,6 +415,10 @@ func (p *Pool) _updateLiquidityAndCrossTick(
 		liquidityNet = new(big.Int).Mul(liquidityNet, constants.NegativeOne)
 	}
 
+	if liquidityNet == nil {
```

Comment:
if `liquidityNet` is nil from line 409 and `willUpTick` is true, I think it will panic on line 415. Is it ok?

Reply:
no, it does not panic, the `liquidityNet` is still nil in line 218
That's why I put this check here, so that we can also have the `newNextTick`